### PR TITLE
BaSP-TG-160-115-191: Create and Update Endpoints - Employees BE

### PR DESCRIPTION
**Pull request by SuarezAlexisEzequiel**
Done:
- TG-160: Create and Update endpoints - Employees BE [Ticket](https://trello.com/c/2mawbiyt/160-create-and-update-endpoints-employees-be)
```
1. Check if endopoints works well
2. Update it if necesary
3. Update middleware
```

- TG-115: BUGS Report: EMPLOYEE [Ticket](https://trello.com/c/dVh07iOt/115-bugs-report-employeel)
```
1. In the <employee> controllers, change the return messages in the json responses. Change "msg" to "message".
2. POST: Change message of create: "Status 200". Create a more specific and clear message. Example: "A new employee has been added successfully."
3. GET by id: Message is uninformative. Create a more specific and clear message. Example: "The employee with the ID has been found."
4. GET ALL: Change message: "Status 200". Create a more specific and clear message. Example: "Employees list displayed correctly."
5. PUT-Update: Change message of update employee: "Status 200". Create a more specific and clear message. Example: "The employee {{_id}} data has been updated correctly."
6. DELETE: Delete "Status 200:" of the success message. The change should be displayed as follows: "Employee with {{_id}} id was deleted". Remember to capitalize the first letter.
7. <CONTROLLERS-EMPLOYEE> Lines: From 45 to 50. Delete all comments.
```

- TG-191: Update EMPLOYEE BE with new data model [Ticket](https://trello.com/c/49axIDUC/191-update-employee-be-with-new-data-model)
```
1. Check that the entity schema follows the new data model. Update if necessary.
2. Modify controllers.
3. Modify validations.
4. Check that the changes work (Postman)
5. Do the changes required on TG-115
```

### CODE REVIEW

File: src/controllers/employees.js [hunk 1]
```diff
@@ -22,70 +23,81 @@ const getEmployeeById = async (req, res) => {
   try {
```
Comment (Lucas-DeLorenzi) [inline src/controllers/employees.js:23]: maybe it's possible to check if the id parameter exists and add the 400 error code option

File: src/controllers/employees.js [hunk 2]
```diff
@@ -22,70 +23,81 @@ const getEmployeeById = async (req, res) => {
...
       error: false,
     });
   } catch (error) {
-    return res.status(500).json({
-      msg: 'Status 500: internal server error',
-      data: undefined,
+    if (firebaseUid) {
+      await Firebase.auth().deleteUser(firebaseUid);
+    }
+    return res.json({
```
Comment (Lucas-DeLorenzi) [inline src/controllers/employees.js:78]: missing error code. A necessary type of error code could be 401

File: src/controllers/employees.js [hunk 3]
```diff
@@ -22,70 +23,81 @@ const getEmployeeById = async (req, res) => {
...
 };
 
 const modifyEmployee = async (req, res) => {
   try {
-    const update = await Employee.findByIdAndUpdate(
-      req.params.id,
-      req.body,
-      { new: true },
-    );
+    const update = await Employee.findByIdAndUpdate(req.params.id, req.body, {
```
Comment (Lucas-DeLorenzi) [inline src/controllers/employees.js:88]: maybe it's possible to check if the id parameter exists and add  400 and 404 error codes options

File: src/validations/employees.js [hunk 4]
```diff
@@ -10,20 +10,25 @@ const validateEmployee = async (req, res, next) => {
     phone: Joi.number().required(),
     email: Joi.string().email().required(),
     password: Joi.string().min(3).max(30).required(),
```
Comment (Lucas-DeLorenzi) [inline src/validations/employees.js:12]: it could have a regex that requires at least one number and one letter

**Review by Lucas-DeLorenzi — CHANGES_REQUESTED** [review]

File: src/validations/employees.js [hunk 5]
```diff
@@ -10,20 +10,25 @@ const validateEmployee = async (req, res, next) => {
...
-    active: Joi.boolean().required(),
   });
+
   const validate = employee.validate(req.body);
   if (validate.error) {
     return res.status(400).json({
+      message: 'There was an error during the validation of the request',
```
Comment (gregorioelias) [inline src/validations/employees.js:18]: Could be added data in the response.

File: src/validations/employees.js [hunk 6]
```diff
@@ -34,20 +39,16 @@ const validateMod = async (req, res, next) => {
     phone: Joi.number(),
     email: Joi.string().email(),
     password: Joi.string().min(3).max(30),
-    active: Joi.boolean(),
   });
+
   const validate = employee.validate(req.body);
   if (validate.error) {
     return res.status(400).json({
+      message: 'There was an error during the validation of the request',
```
Comment (gregorioelias) [inline src/validations/employees.js:47]: Could be added data in the response.

**Review by Lucas-DeLorenzi — APPROVED** [review]

**Review by gregorioelias — APPROVED** [review]

**Pull request merged**